src/Microsoft.AspNetCore.Http.Extensions/UriHelper.cs

```diff
@@ -80,7 +80,7 @@ public static string Encode(Uri uri)
         {
             if (uri.IsAbsoluteUri)
             {
-                return Encode(
+                return BuildAbsolute(
```

Comment:
System.Uri has minimal support for relative uris. It won't parse it into path?query#fragment for you. The most it will do is some basic escaping.